GVRf/Framework/jni/objects/mesh.h

```diff
@@ -259,6 +296,21 @@ class Mesh: public HybridObject {
 
     const BoundingVolume& getBoundingVolume();
 
+    bool hasBones() const {
+        return vertexBoneData_.getNumBones();
+    }
+
+    void setBones(std::vector<Bone*>&& bones) {
+        vertexBoneData_.setBones(std::move(bones));
+        shouldResetBoneBuffers = true;
+    }
+
+    VertexBoneData &getVertexBoneData() {
+        return vertexBoneData_;
+    }
+
+    void generateBoneArrayBuffers(Material::ShaderType key);
```

Comment:
ShaderType key can be removed if it is not in the base code. I don't use it to distinguish meshes generated.